packages/lib/gpcircuits/circuits/proto-pod-gpc.circom

```diff
@@ -188,29 +197,69 @@ template ProtoPODGPC (
     }
 
     /*
-     * 1 OwnerModule with its inputs & outputs.
+     * External nullifier for owner modules (if any).
+     *
+     * The final nullifier hash(es) will be calculated based on this
+     * and the owner's identity.
+     */
+    signal input ownerExternalNullifier;
+    
+    /*
+     * <=1 OwnerModule with its inputs & outputs.
      */
 
     // Entry containing owner's Semaphore V3 commitment (public), or -1 to disable ownership checking.
-    signal input ownerEntryIndex;
+    signal input ownerEntryIndex[INCLUDE_OWNERV3];
 
     // Owner's Semaphore V3 identity (private key) kept hidden and verified.
-    signal input ownerSemaphoreV3IdentityNullifier, ownerSemaphoreV3IdentityTrapdoor;
+    signal input ownerSemaphoreV3IdentityNullifier[INCLUDE_OWNERV3], ownerSemaphoreV3IdentityTrapdoor[INCLUDE_OWNERV3];
 
-    // Final nullifier hash is calculated based on external nullifier and owner's identity.
-    signal input ownerExternalNullifier, ownerIsNullfierHashRevealed;
+    // Indicator of whether the nullifier hash should be revealed.
+    signal input ownerIsNullifierHashRevealed[INCLUDE_OWNERV3];
 
     // Owner module verifies owner's ID, and generates nullifier.
-    signal ownerIsEnabled <== NOT()(IsZero()(ownerEntryIndex + 1));
-    signal output ownerRevealedNullifierHash <== OwnerModuleSemaphoreV3()(
-        enabled <== ownerIsEnabled,
-        identityNullifier <== ownerSemaphoreV3IdentityNullifier,
-        identityTrapdoor <== ownerSemaphoreV3IdentityTrapdoor,
-        identityCommitmentHash <== InputSelector(MAX_ENTRIES)(entryValueHashes, ownerIsEnabled * ownerEntryIndex),
-        externalNullifier <== ownerExternalNullifier,
-        isNullfierHashRevealed <== ownerIsNullfierHashRevealed
-                                                                          );
+    signal output ownerRevealedNullifierHash[INCLUDE_OWNERV3];
+    signal ownerIsEnabled[INCLUDE_OWNERV3];
+        
+    for (var i = 0; i < INCLUDE_OWNERV3; i++) {
+        ownerIsEnabled[i] <== NOT()(IsZero()(ownerEntryIndex[i] + 1));
+        ownerRevealedNullifierHash[i] <== OwnerModuleSemaphoreV3()(
+            enabled <== ownerIsEnabled[i],
+            identityNullifier <== ownerSemaphoreV3IdentityNullifier[i],
+            identityTrapdoor <== ownerSemaphoreV3IdentityTrapdoor[i],
+            identityCommitmentHash <== InputSelector(MAX_ENTRIES)(entryValueHashes, ownerIsEnabled[i] * ownerEntryIndex[i]),
+            externalNullifier <== ownerExternalNullifier,
+            isNullifierHashRevealed <== ownerIsNullifierHashRevealed[i]
+                                                                   );
+    }
+    /*
+     * <=1 OwnerModuleV4 with its inputs & outputs.
```

Comment:
The two owner blocks are pretty much the same, so assume all my comments above apply here if relevant.
I wonder if there's a way to make this code reusable so there wasn't so much duplication.
In C++ I'd use a macro and instantiate it twice for 3 and 4.  Without that I'm not sure if there's a form of generic programming in circom which is powerful enough.  I think you could make an OwnerModule(V) which could be instantiated as OwnerModule(3) or OwnerModule(4) with an internal if statement to pick the underlying module.  However there would be almost as much boiler-plate just to pass in all of the named inputs from this level, so I'd have to see the code to be sure if it really seems better.  Feel free to experiment with that if you think it's worthwhile.